micro/pattern_shared.cpp

```diff
@@ -34,7 +34,7 @@ class MicroBenchLocalMemory
       auto out = output_buf.template get_access<s::access::mode::discard_write>(cgh);
       // local memory definition
       s::accessor<DATA_TYPE, 1,s::access::mode::read_write, s::access::target::local> 
-        local_mem(s::range<1>(TILE_DIM), cgh);
+        local_mem(TILE_DIM, cgh);
```

Comment:
ComputeCpp really complains about that? The signature of the local memory accessor constructor according to spec is clearly `accessor(range<dimensions> allocationSize, handler &commandGroupHandlerRef)`. Is this a ComputeCpp bug?

Reply:
Honestly I have no idea what's going on here, the error I get is `error: declaration of ‘cl::sycl::range<1> TILE_DIM’ shadows template parameter` -- which doesn't make any sense to me.

Reply:
It might be some weird dependent-name two-phase parsing issue..?

Reply:
This happens with hipSYCL on CPU as well when using gcc as host compiler. I assume ComputeCpp also relies on gcc here. Maybe we should just rename `TILE_DIM` to resolve the issue once and for all?

Reply:
Yes, this seems to happen during host-compilation which uses GCC by default on my machine. The fact that the device compiler (which is based on Clang) doesn't complain leads me to believe this might be a GCC bug. Anyhow, it's not actually related to the name of the parameter at all, we can either rely on the conversion constructor of `range<1>`, as I'm doing now, or use uniform initialization syntax, which also works.

Reply:
Ah sorry, I now see that there's really only one definition of `TILE_DIM`. When I encountered this issue I assumed that the compiler knows what it's doing, so without checking I just assumed there's probably a second definition somewhere... But you're right, this here doesn't make sense at all, so probably weird compiler bug. Let's just rely on your solution and try to forget this weirdness ;)